api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -136,21 +141,15 @@ public void syncBillingProjectStatus() {
   }
 
   public void cleanBillingBuffer() {
+    Instant now = clock.instant();
     Iterables.concat(
-            billingProjectBufferEntryDao.findAllByStatusAndLastStatusChangedTimeLessThan(
-                DbStorageEnums.billingProjectBufferStatusToStorage(CREATING),
-                new Timestamp(
-                    clock
-                        .instant()
-                        .minus(CREATING_TIMEOUT_MINUTES, ChronoUnit.MINUTES)
-                        .toEpochMilli())),
+            findExpiredCreatingEntries(now),
+            // For the ASSIGNING status monitor, we can simply filter by the current time as this
+            // status tracks internal state rather than the Firecloud status.
             billingProjectBufferEntryDao.findAllByStatusAndLastStatusChangedTimeLessThan(
```

Comment:
nit: Aesthetically I might change "TimeLessThan" to "EarlierThan" in this name. Just because I don't think of wall clock time as something comparable in that way.